### PR TITLE
docs: Reviewed and modified the new contents + modified the icons in the sidebar

### PR DESCRIPTION
**Pull request by balaharish7**

### CODE REVIEW

**Comment by balaharish7** [comment]
14-Mar-24: @avishnu @Abhinandan-Purkait @niladrih @datacore-gthomas 

We have modified & the reviewed the 1st two sections (Introduction to OpenEBS & Concepts).
Kindly verify and share your feedback, if any by EOB 15-Mar-24. :-)

**Comment by orville-wright** [comment]
Looks good. I personally think we should not repeat the word "Guide" everywhere in the tree view (if possible).

**Comment by balaharish7** [comment]
We have modified & the reviewed the Troubleshooting section.
Kindly verify and share your feedback.

File: docs/i18n/en/docusaurus-plugin-content-docs/version-3.0.x.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{
```
Comment (avishnu) [inline docs/i18n/en/docusaurus-plugin-content-docs/version-3.0.x.json:1]: why is this 3.0.x.json?

File: docs/main/concepts/architecture.md [hunk 2]
```diff
@@ -52,21 +52,20 @@ The implementation pattern used by data engines to provide high availability is
...
-- Device or Directory paths in case of Local PV, 
-- iSCSI Target in case of cStor and Jiva
-- NVMe Target in case of Mayastor.
+- Device or Directory paths in case of Local Engine
+- NVMe Target in case of Replicated Engine
 
 ### Volume Data Layer 
 
-OpenEBS Data engines create a Volume Replica on top of the Storage Layer. Volume Replicas are pinned to a node and are created on top of the storage layer. The replica can be any of the following:
+OpenEBS Data Engines create a Volume Replica on top of the Storage Layer. Volume Replicas are pinned to a node and are created on top of the Storage Layer. The replica can be any of the following:
```
Comment (avishnu) [inline docs/main/concepts/architecture.md:60]: storage layer case-change not needed IMO

File: docs/main/concepts/architecture.md [hunk 3]
```diff
@@ -52,21 +52,20 @@ The implementation pattern used by data engines to provide high availability is
...
 
 - Sub-directory - in case the storage layer used is a filesystem directory
 - Full Device or Partitioned Device - in case the storage layer used is block devices
```
Comment (avishnu) [inline docs/main/concepts/architecture.md:63]: This line needs to be deleted.

File: docs/main/concepts/architecture.md [hunk 4]
```diff
@@ -133,25 +132,26 @@ CSI Drivers act as the facilitators for managing the life-cycle of volumes withi
...
-- Visualizations  (provided via standard Kubernetes Dashboards or custom Grafana dashboards)
+- Monitoring and Alerting (provided via integrations into Prometheus, Grafana, Alert manager) 
+- Enforcing Security Policies (provided via integrations with PodSecurityPolicies or Kyerno) 
+- Logging (provide via integration to any standard Logging stack setup by administrators like ELK, Loki, Logstash) 
+- Visualizations (provided via standard Kubernetes Dashboards or custom Grafana dashboards)
+
 ### CLI 
 
-All the management functions on OpenEBS can be carried out via `kubectl` as OpenEBS uses Custom Resources for managing all of its configurations and reporting the status of the components. 
+All the management functions on OpenEBS can be carried out via `kubectl` as OpenEBS uses custom resources for managing all of its configurations and reporting the status of the components.
```
Comment (avishnu) [inline docs/main/concepts/architecture.md:151]: Retain Custom Resources

File: docs/main/concepts/basics.md [hunk 5]
```diff
@@ -13,10 +13,10 @@ description: This document provides you with a quick overview of the Kubernetes
 ---
 
 :::note
-This page provides you with a quick overview of the [Kubernetes concepts](https://kubernetes.io/docs/concepts/storage/) you need to know for running Stateful Workloads. If you are already familiar with running Stateful workloads in Kubernetes, header over to the next section on [Container Attached Storage](/docs/concepts/cas).
+This page provides you with a quick overview of the [Kubernetes concepts](https://kubernetes.io/docs/concepts/storage/) you need to know for running Stateful Workloads. If you are already familiar with running Stateful workloads in Kubernetes, header over to the next section on[Container Native Storage](container-native-storage.md).
```
Comment (avishnu) [inline docs/main/concepts/basics.md:16]: Need a space after "next section on"

File: docs/main/concepts/basics.md [hunk 6]
```diff
@@ -13,10 +13,10 @@ description: This document provides you with a quick overview of the Kubernetes
...
+Kubernetes has made several enhancements to support running Stateful eorkloads by providing the required abstractions for platform (or Cluster Administrators) and application developers. The abstractions ensure that different types of file and block storage (whether ephemeral or persistent, local or remote) are available wherever a container is scheduled (including provisioning/creating, attaching, mounting, unmounting, detaching, and deleting of volumes), storage capacity management (container ephemeral storage usage, volume resizing, etc.), influencing scheduling of containers based on storage (data gravity, availability, etc.), and generic operations on storage (snapshotting, etc.).
```
Comment (avishnu) [inline docs/main/concepts/basics.md:19]: Typo - workloads

File: docs/main/concepts/basics.md [hunk 7]
```diff
@@ -33,36 +33,36 @@ The [Container Storage Interface (CSI)](https://github.com/container-storage-int
...
 
-The dynamic provisioning feature eliminates the need for cluster administrators to pre-provision storage. Instead, it automatically provisions storage when it is requested by users. The implementation of dynamic volume provisioning is based on the `StorageClass` abstraction A cluster administrator can define as many StorageClass objects as needed, each specifying a volume plugin (aka provisioner) that provisions a volume and the set of parameters to pass to that provisioner when provisioning. 
+The dynamic provisioning feature eliminates the need for cluster administrators to pre-provision storage. Instead, it automatically provisions storage when it is requested by users. The implementation of dynamic volume provisioning is based on the `StorageClass` abstraction. A cluster administrator can define as many storage class objects as needed, each specifying a volume plugin (aka provisioner) that provisions a volume and the set of parameters to pass to that provisioner when provisioning.
```
Comment (avishnu) [inline docs/main/concepts/basics.md:43]: Retain StorageClass

File: docs/main/concepts/basics.md [hunk 8]
```diff
@@ -33,36 +33,36 @@ The [Container Storage Interface (CSI)](https://github.com/container-storage-int
...
+When OpenEBS is installed, it ships with a couple of default storage classes that allow users to create either local or replicated volumes. The cluster administrator can enable the required storage engines and then create storage classes for the required Data Engines. 
 
 ## Persistent Volume Claims 
 
-PersistentVolumeClaim (PVC) is a user’s storage request that is served by a StorageClass offered by the cluster administrator. An application running on a container can request a certain type of storage. For example, a container can specify the size of storage it needs or the way it needs to access the data (read only, read/write, read-write many, etc., ).
+Persistent Volume Claim (PVC) is a user’s storage request that is served by a storage class offered by the cluster administrator. An application running on a container can request a certain type of storage. For example, a container can specify the size of storage it needs or the way it needs to access the data (read only, read/write, etc.).
```
Comment (avishnu) [inline docs/main/concepts/basics.md:51]: PersistentVolumeClaim is a single word, StorageClass is also a single word in this sentence.

File: docs/main/concepts/basics.md [hunk 9]
```diff
@@ -33,36 +33,36 @@ The [Container Storage Interface (CSI)](https://github.com/container-storage-int
...
+Beyond storage size and access mode, administrators create storage classes to provided PVs with custom properties, such as the type of disk (HDD vs. SSD), the level of performance, or the storage tier (regular or cold storage). 
 
 ## Persistent Volumes
 
-The PersistentVolume(PV) is dynamically provisioned by the storage providers when users request for a PVC. PV contains the details on how the storage can be consumed by the container. Kubernetes and the Volume Drivers use the details in the PV to attach/detach the storage to the node where the container is running and mount/unmount storage to a container. 
+The Persistent Volume (PV) is dynamically provisioned by the storage providers when users request for a PVC. PV contains the details on how the storage can be consumed by the container. Kubernetes and the volume drivers use the details in the PV to attach/detach the storage to the node where the container is running and mount/unmount storage to a container.
```
Comment (avishnu) [inline docs/main/concepts/basics.md:57]: PersistentVolume is also a single word

File: docs/main/concepts/container-native-storage.md [hunk 10]
```diff
@@ -1,49 +1,51 @@
...
 
-CAS reflects a broader trend of solutions – many of which are now part of Cloud Native Foundation – that reinvent particular categories or create new ones – by being built on Kubernetes and microservice and that deliver capabilities to Kubernetes based microservice environments. For example, new projects for security, DNS, networking, network policy management, messaging, tracing, logging and more have emerged in the cloud-native ecosystem and often in CNCF itself.
+CNS reflects a broader trend of solutions – many of which are now part of Cloud Native Computing Foundation (CNCF) – that reinvent particular categories or create new ones – by being built on Kubernetes and microservice and that deliver capabilities to Kubernetes based microservice environments. For example, new projects for security, DNS, networking, network policy management, messaging, tracing, logging, and more have emerged in the cloud-native ecosystem and often in CNCF itself.
```
Comment (avishnu) [inline docs/main/concepts/container-native-storage.md:20]: Good !!

File: docs/main/concepts/data-engines/local-engine.md [hunk 11]
```diff
@@ -1,52 +1,46 @@
...
-- Edge workloads that typically run on a single node or in Single node Kubernetes Clusters.
+- Edge workloads that typically run on a single node or in single node Kubernetes Clusters.
 
 OpenEBS helps users to take local volumes into production by providing features that are currently missing in Kubernetes like:
 
-- Dynamic PV Provisioners for local volumes.
-- Local Volumes backed by hostpath on filesystems like Ext3, XFS, LVM or ZFS.
-- Monitoring the health of underlying devices or storage used to create Local Volumes. 
+- Dynamic PV provisioners for local volumes.
+- Local volumes backed by hostpath on filesystems like Ext3, XFS.
```
Comment (avishnu) [inline docs/main/concepts/data-engines/local-engine.md:27]: No need to remove "LVM or ZFS"

File: docs/main/concepts/data-engines/local-engine.md [hunk 12]
```diff
@@ -1,52 +1,46 @@
...
+- Monitoring the health of underlying devices or storage used to create local volumes. 
 - Capacity management features like over-provisioning and/or quota enforcement. 
-- Make use of the underlying storage capabilities like snapshot, clone, compression and so forth when local volumes are backed by advanced filesystem like ZFS. 
+- Make use of the underlying storage capabilities like snapshot, clone, compression, and so forth when local volumes are backed by advanced filesystem.
```
Comment (avishnu) [inline docs/main/concepts/data-engines/local-engine.md:30]: like ZFS and LVM.

File: docs/main/troubleshooting/troubleshooting-replicated-engine.md [hunk 13]
```diff
@@ -1,5 +1,5 @@
 ---
-id: troubleshooting-re
+id: troubleshootingre
```
Comment (avishnu) [inline docs/main/troubleshooting/troubleshooting-replicated-engine.md:2]: Why this change?

File: docs/sidebars.js [hunk 14]
```diff
@@ -366,20 +355,34 @@ module.exports = {
         },
         {
           type: "doc",
-          id: "troubleshooting/troubleshooting-re",
+          id: "troubleshooting/troubleshootingre",
```
Comment (avishnu) [inline docs/sidebars.js:358]: why this change?

File: docs/i18n/en/code.json [hunk 15]
```diff
@@ -193,15 +193,15 @@
   },
   "theme.docs.versions.unreleasedVersionLabel": {
     "message": "This is the documentation for the development version of OpenEBS.",
-    "description": "The label used to tell the user that they are browsing an unreleased doc version"
+    "description": "The label used to tell the user that he's browsing an unreleased doc version"
```
[the remark this answers is not in the document]
Comment (balaharish7) [reply docs/i18n/en/code.json:196]: We didn't change this & we're not sure how this got changed as we didn't work on this page. Anyways, we've modified the change now.

File: docs/i18n/en/docusaurus-plugin-content-docs/version-3.0.x.json (hunk 1 above)
Comment (balaharish7) [reply docs/i18n/en/docusaurus-plugin-content-docs/version-3.0.x.json:1]: We didn't change this & we're not sure what version should it be. Kindly let us know, what's the correct version, so that we can modify this as well.

File: docs/main/concepts/architecture.md (hunk 2 above)
Comment (balaharish7) [reply docs/main/concepts/architecture.md:60]: This has been modified now. :)

File: docs/main/concepts/architecture.md (hunk 3 above)
Comment (balaharish7) [reply docs/main/concepts/architecture.md:63]: We have few bullet points and other related contents below the highlighted line. Should we remove this as well?

```
**OpenEBS Data Engines create a Volume Replica on top of the storage layer. Volume Replicas are pinned to a node and are created on top of the storage layer. The replica can be any of the following:

- Sub-directory - in case the storage layer used is a filesystem directory
- Full Device or Partitioned Device - in case the storage layer used is block devices
- Logical Volume - in case the storage layer used is a device pool coming from local engine

In case the applications require only local storage, then the persistent volume will be created using one of the above directories, device (or partition) or logical volume. OpenEBS [control plane](#control-plane) will be used to provision one of the above replicas.** 
```

**Comment by balaharish7** [comment]
19-Mar-24: @avishnu @Abhinandan-Purkait @niladrih @datacore-gthomas

We have modified & the reviewed the below sections:

- QuickStart Guide
      - QuickStart
      - Installation
      - Deploy a Test Application
- FAQs
- Community
- Commercial Support

_Note: Deploy a Test Application section contains the Mayastor Configuring docs as well. We need the contents for Deploy a Test Application & Configuration of Local Engine.
Based on our review, we would suggest we can have a separate page for Deployment and a separate page for Configuration._

Kindly verify and share your feedback.

File: docs/i18n/en/code.json [hunk 16]
```diff
@@ -197,11 +197,11 @@
   },
   "theme.docs.versions.unmaintainedVersionLabel": {
     "message": "This is documentation for {siteTitle} {versionLabel}, which is no longer actively maintained.",
-    "description": "The label used to tell the user that they are browsing an unmaintained doc version"
+    "description": "The label used to tell the user that he's browsing an unmaintained doc version"
```
Comment (niladrih) [inline docs/i18n/en/code.json:200]: Oh, we can't do this I think.

Comment (balaharish7) [reply docs/i18n/en/code.json:200]: Yeah! We didn't do this Niladri. We never worked on this document & we both (myself & Gincy) are not sure how this got changed.

**Comment by balaharish7** [comment]
20-Mar-24: @avishnu @Abhinandan-Purkait @niladrih @datacore-gthomas

We have modified & the reviewed the below sections:

- Local Engine Prerequisites
- Local Engine Installation

Kindly verify and share your feedback.

File: docs/i18n/en/code.json (hunk 15 above)
[the remark this answers is not in the document]
Comment (avishnu) [reply docs/i18n/en/code.json:196]: I still see "he" and not "they"

File: docs/i18n/en/docusaurus-plugin-content-docs/version-3.0.x.json (hunk 1 above)
Comment (avishnu) [reply docs/i18n/en/docusaurus-plugin-content-docs/version-3.0.x.json:1]: why is this file created newly?
Similarly for other files of this type.

File: docs/main/concepts/architecture.md [hunk 17]
```diff
@@ -101,18 +100,18 @@ OpenEBS Control Plane comprises of a set of micro-services that are themselves m
...
 
-As part of the OpenEBS install, the control plane components for the selected data engines will be installed as cluster and/or node components using standard Kubernetes primitives like Deployments, DaemonSets, Statefulsets and such. The OpenEBS installation also takes care of loading the OpenEBS custom resource definitions into the Kubernetes. 
+As part of the OpenEBS install, the control plane components for the selected data engines will be installed as cluster and/or node components using standard Kubernetes primitives like Deployments, DaemonSets, Statefulsets, and such. The OpenEBS installation also takes care of loading the OpenEBS custom resource definitions into the Kubernetes. 
 
-OpenEBS control plane components are all stateless and depend on the Kubernetes etcd server (custom resources) to managing their internal configuration state and reporting the status of the various components.  
+OpenEBS control plane components are all stateless and depend on the Kubernetes etcd server (custom resources) to managing their internal configuration state and reporting the status of the various components.
```
Comment (avishnu) [inline docs/main/concepts/architecture.md:105]: change this to "depend on the custom resources and etcd server for managing..."

File: docs/main/concepts/architecture.md [hunk 18]
```diff
@@ -101,18 +100,18 @@ OpenEBS Control Plane comprises of a set of micro-services that are themselves m
...
 
 ### Declarative API 
 
 OpenEBS supports Declarative API for managing all of its operations and the APIs are exposed as Kubernetes custom resources. Kubernetes CRD validators and admission webhooks are used to validate the inputs provided by the users and to validate if the operations are allowed.  
 
 The Declarative API is a natural extension to what Kubernetes administrators and user are accustomed to, where they can define the intent via a YAML and then Kubernetes and associated OpenEBS Operators will reconcile the state with the user's intent. 
 
-The Declarative API can be used to configure the Data Engines and setup volume profiles/policies.  Even upgrades of the data engines are performed using this API. The API can be used to: 
-- Manage the configuration for each of the Data engine
+The Declarative API can be used to configure the Data Engines and setup volume profiles/policies. Even upgrades of the data engines are performed using this API. The API can be used to:
```
Comment (avishnu) [inline docs/main/concepts/architecture.md:113]: each Data engine

File: docs/main/concepts/architecture.md [hunk 19]
```diff
@@ -101,18 +100,18 @@ OpenEBS Control Plane comprises of a set of micro-services that are themselves m
...
 ### Declarative API 
 
 OpenEBS supports Declarative API for managing all of its operations and the APIs are exposed as Kubernetes custom resources. Kubernetes CRD validators and admission webhooks are used to validate the inputs provided by the users and to validate if the operations are allowed.  
 
 The Declarative API is a natural extension to what Kubernetes administrators and user are accustomed to, where they can define the intent via a YAML and then Kubernetes and associated OpenEBS Operators will reconcile the state with the user's intent. 
 
-The Declarative API can be used to configure the Data Engines and setup volume profiles/policies.  Even upgrades of the data engines are performed using this API. The API can be used to: 
-- Manage the configuration for each of the Data engine
+The Declarative API can be used to configure the Data Engines and setup volume profiles/policies. Even upgrades of the data engines are performed using this API. The API can be used to: 
+- Manage the configuration for each of the Data Engine
```
Comment (avishnu) [inline docs/main/concepts/architecture.md:114]: each data engine

File: docs/main/concepts/basics.md [hunk 20]
```diff
@@ -13,10 +13,10 @@ description: This document provides you with a quick overview of the Kubernetes
 ---
 
 :::note
-This page provides you with a quick overview of the [Kubernetes concepts](https://kubernetes.io/docs/concepts/storage/) you need to know for running Stateful Workloads. If you are already familiar with running Stateful workloads in Kubernetes, header over to the next section on [Container Attached Storage](/docs/concepts/cas).
+This page provides you with a quick overview of the [Kubernetes concepts](https://kubernetes.io/docs/concepts/storage/) you need to know for running Stateful Workloads. If you are already familiar with running Stateful workloads in Kubernetes, header over to the next section on [Container Native Storage](container-native-storage.md).
```
Comment (avishnu) [inline docs/main/concepts/basics.md:16]: Is the path to the md correct?

File: docs/main/quickstart-guide/quickstart.md [hunk 21]
```diff
@@ -44,87 +39,82 @@ The OpenEBS workflow fits nicely into the reconcilation pattern introduced by Ku
...
+- [RHEL](../user-guides/local-engine-user-guide/prerequisites.mdx)
+- [CentOS](../user-guides/local-engine-user-guide/prerequisites.mdx)
+- [OpenShift](../user-guides/local-engine-user-guide/prerequisites.mdx)
+- [Rancher](../user-guides/local-engine-user-guide/prerequisites.mdx)
+- [ICP](../user-guides/local-engine-user-guide/prerequisites.mdx)
+- [EKS](../user-guides/local-engine-user-guide/prerequisites.mdx)
+- [GKE](../user-guides/local-engine-user-guide/prerequisites.mdx)
+- [AKS](../user-guides/local-engine-user-guide/prerequisites.mdx)
+- [Digital Ocean](../user-guides/local-engine-user-guide/prerequisites.mdx)
+- [Konvoy](../user-guides/local-engine-user-guide/prerequisites.mdx)
```
Comment (avishnu) [inline docs/main/quickstart-guide/quickstart.md:65]: Did we miss the positional reference with #?

File: docs/i18n/en/docusaurus-plugin-content-docs/version-3.0.x.json (hunk 1 above)
Comment (balaharish7) [reply docs/i18n/en/docusaurus-plugin-content-docs/version-3.0.x.json:1]: We didn't create this file @avishnu.

File: docs/main/troubleshooting/troubleshooting-replicated-engine.md (hunk 13 above)
Comment (balaharish7) [reply docs/main/troubleshooting/troubleshooting-replicated-engine.md:2]: The ID "**troubleshooting**" already exists for _Troubleshooting page of Local Engine_. hence, we've created a new ID "**troubleshootingre**" for the _Troubleshooting page of Replicated Engine_.

File: docs/sidebars.js (hunk 14 above)
Comment (balaharish7) [reply docs/sidebars.js:379]: The ID "troubleshooting" already exists for Troubleshooting page of Local Engine. hence, we've created a new ID "troubleshootingre" for the Troubleshooting page of Replicated Engine.

File: docs/main/concepts/architecture.md (hunk 17 above)
Comment (balaharish7) [reply docs/main/concepts/architecture.md:105]: This is now modified :)

File: docs/main/concepts/architecture.md (hunk 19 above)
Comment (balaharish7) [reply docs/main/concepts/architecture.md:114]: This change is done now 👍

File: docs/main/concepts/basics.md (hunk 20 above)
Comment (balaharish7) [reply docs/main/concepts/basics.md:16]: Yes! The path is correct. 👍

File: docs/main/quickstart-guide/quickstart.md (hunk 21 above)
Comment (balaharish7) [reply docs/main/quickstart-guide/quickstart.md:65]: No Vishnu. We didn't miss the positional references. All these sections were talking about **iSCSI** which is not applicable now and those sections are removed now. Hence, the positional references are not made.

Also, now the "Prerequisites" page won't be there in the new structure as we're planning to have separate Prerequisites for each engine.

File: docs/main/concepts/architecture.md [hunk 22]
```diff
@@ -91,39 +90,39 @@ The control plane in the context of OpenEBS refers to a set of tools or componen
...
 - Integrating into other tools like Prometheus/Grafana for telemetry and monitoring 
-- Integrating into other tools for debugging, troubleshooting or log management
+- Integrating into other tools for debugging, troubleshooting, or log management
 
 OpenEBS Control Plane comprises of a set of micro-services that are themselves managed by Kubernetes, making OpenEBS truly Kubernetes native. The configuration managed by the OpenEBS Control Plane is saved as Kubernetes custom resources. The functionality of the control plane can be decomposed into the various stages as follows:
 
 ![openebs control plane](../assets/openebs-control-plane.svg)
 
-### YAML or Helm chart
+### YAML or Helm Chart
```
Comment (avishnu) [inline docs/main/concepts/architecture.md:99]: We only support helm chart based installation.

**Review by avishnu — APPROVED** [review]
Looks good at the outset. Need to validate once on the live website.
